### ISSUE TITLE
When using Sails-disk and the default schema false, migrate alter will remove attributes not in the schema

### ISSUE THREAD

**Issue by irlnathan** [balderdashy/waterline#966]

**Comment by tjwebb** [comment]
Yikes. Good thing I just merged https://github.com/balderdashy/waterline/pull/962 :)

**Comment by particlebanana** [comment]
@tjwebb haven't tested it but I suspect it would also be an issue with sails-mongo and other nosql dbs that have `schema: false` set by default.

Also just as a thought I think #962 will mess with sails projects. Unless Sails is overriding it and setting `migrate: 'alter'` somewhere manually.

**Comment by devinivy** [comment]
Possibly related: https://github.com/balderdashy/waterline/pull/959

**Comment by dmarcelino** [comment]
#959 has been released to npm yet though, so it could only cause this if @irlnathan is using waterline github version.

**Comment by dmarcelino** [comment]
@particlebanana, we can easily postpone #962 to `0.11`.

**Comment by particlebanana** [comment]
@dmarcelino we might want to wait on it. Unless there is somewhere else in Sails that sets it, default projects that get generated have the `migrate` config commented out by default. If this was a patch it would break the normal behavior of Sails. We should take a look and fix that in Sails if it's the case.

From the looks of it I think #959 solves this issue. If we revert #962 are we good for a publish?

**Comment by dmarcelino** [comment]
Yeap, given you gave your :+1:  to #959 we are good to release waterline. I'll revert #962 and issue a PR against `0.11`.

**Comment by dmarcelino** [comment]
@particlebanana, I've issued #967 to revert #962.

**Comment by dmarcelino** [comment]
@irlnathan, can you check if this is still broken with the latest code in master? PR #959 should have fix this.

**Comment by particlebanana** [comment]
Issue seems to still exist. Going to look at getting a failing test created.

**Comment by dmarcelino** [comment]
I see. If you need inspiration for the test, you can get some from @khalilTN's PR #1005 which adds a test for `migrate: 'alter'`.
